Readme.md

```diff
@@ -53,8 +53,27 @@ Parallel.all?([1,2,nil,4,5]) { |number| number != nil }
 # => false
 ```
 
+To avoid overhead from reducing big result sets at the end of processing, use `reduce` this method will return results collected from each worker. You need to merge them on the end to get the fully reduced result.
 
-Processes/Threads are workers, they grab the next piece of work when they finish.
```

Comment:
only document 1 way of doing things, this is already complicated enough and will only be needed by very few users

Reply:
naming it `reduce` suggests that it will return 1 result, but it does not, also we might later add an actual `reduce`
so reduce_by_worker might be more clear

Reply:
Sure